system/lib/libc/emscripten_syscall_stubs.c

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright 2021 The Emscripten Authors.  All rights reserved.
+ * Emscripten is available under two separate licenses, the MIT license and the
+ * University of Illinois/NCSA Open Source License.  Both these licenses can be
+ * found in the LICENSE file.
+ *
+ * Unimplemented/dummy syscall implemenations. These fall into 3 catagories.
+ *
+ * 1. Fake it, use dummy/placeholder values and return success.
+ * 2. Fake it, as above but warn at runtime if called.
+ * 3. Return ENOSYS and warn at runtime if called.
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <syscall_arch.h>
+#include <string.h>
+#include <sys/resource.h>
+#include <sys/stat.h>
+#include <time.h>
+#include <sys/utsname.h>
+
+static int g_pid = 42;
+static int g_pgid = 42;
+static int g_ppid = 1;
+static int g_sid = 42;
+static mode_t g_umask = S_IRWXU | S_IRWXG | S_IRWXO;
+
+#ifdef NDEBUG
+#define REPORT(name)
+#else
+#define REPORT(name) \
+  fprintf(stderr, "warning: unsupported syscall: __syscall_" #name "\n");
+#endif
+
+#define UNIMPLEMNTED(name, args) \
+  long __syscall_##name args { \
+    REPORT(name); \
+    return -ENOSYS; \
+  }
+
+long __syscall_uname(long buf) {
+  if (!buf) {
+    return -EFAULT;
+  }
+  struct utsname *utsname = (struct utsname *)buf;
+
+  strcpy(utsname->sysname, "Emscripten");
+  strcpy(utsname->nodename, "emscripten");
+  strcpy(utsname->release, "1.0");
```

Comment:
Do we have a macro for the emscripten version?
Or I guess we could do sprintf with EMSCRIPTEN_MAJOR et al?

Reply:
I'm just moving the code from JS to native.    Better not to change the implementation at the same time I think.